## Some context

 - dataset contains transactions from September 2013 in Europe
 - highly unbalanced
 - features V_1, V_2, ..., V_28 obtained from PCA
 - feature 'Time' = delta of time elapsed from the first transaction

In [1]:
# libraries
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
pd.set_option('display.float_format', lambda x: '%.3f' % x)  # using a 3 decimal format
df = pd.read_csv("../creditcard.csv")  # read the data
df.head()

Time     V1     V2    V3     V4     V5     V6     V7     V8     V9  ...  \
0 0.000 -1.360 -0.073 2.536  1.378 -0.338  0.462  0.240  0.099  0.364  ...   
1 0.000  1.192  0.266 0.166  0.448  0.060 -0.082 -0.079  0.085 -0.255  ...   
2 1.000 -1.358 -1.340 1.773  0.380 -0.503  1.800  0.791  0.248 -1.515  ...   
3 1.000 -0.966 -0.185 1.793 -0.863 -0.010  1.247  0.238  0.377 -1.387  ...   
4 2.000 -1.158  0.878 1.549  0.403 -0.407  0.096  0.593 -0.271  0.818  ...   

     V21    V22    V23    V24    V25    V26    V27    V28  Amount  Class  
0 -0.018  0.278 -0.110  0.067  0.129 -0.189  0.134 -0.021 149.620      0  
1 -0.226 -0.639  0.101 -0.340  0.167  0.126 -0.009  0.015   2.690      0  
2  0.248  0.772  0.909 -0.689 -0.328 -0.139 -0.055 -0.060 378.660      0  
3 -0.108  0.005 -0.190 -1.176  0.647 -0.222  0.063  0.061 123.500      0  
4 -0.009  0.798 -0.137  0.141 -0.206  0.502  0.219  0.215  69.990      0  

[5 rows x 31 columns]

We can observe that V_1 through V_28 have been scaled

In [3]:
df.describe()

Time         V1         V2         V3         V4         V5  \
count 284807.000 284807.000 284807.000 284807.000 284807.000 284807.000   
mean   94813.860      0.000      0.000     -0.000      0.000      0.000   
std    47488.146      1.959      1.651      1.516      1.416      1.380   
min        0.000    -56.408    -72.716    -48.326     -5.683   -113.743   
25%    54201.500     -0.920     -0.599     -0.890     -0.849     -0.692   
50%    84692.000      0.018      0.065      0.180     -0.020     -0.054   
75%   139320.500      1.316      0.804      1.027      0.743      0.612   
max   172792.000      2.455     22.058      9.383     16.875     34.802   

              V6         V7         V8         V9  ...        V21        V22  \
count 284807.000 284807.000 284807.000 284807.000  ... 284807.000 284807.000   
mean       0.000     -0.000      0.000     -0.000  ...      0.000     -0.000   
std        1.332      1.237      1.194      1.099  ...      0.735      0.726   
min      -26.161    -43.557    -73.217    -13.434  ...    -34.830    -10.933   
25%       -0.768     -0.554     -0.209     -0.643  ...     -0.228     -0.542   
50%       -0.274      0.040      0.022     -0.051  ...     -0.029      0.007   
75%        0.399      0.570      0.327      0.597  ...      0.186      0.529   
max       73.302    120.589     20.007     15.595  ...     27.203     10.503   

             V23        V24        V25        V26        V27        V28  \
count 284807.000 284807.000 284807.000 284807.000 284807.000 284807.000   
mean       0.000      0.000      0.000      0.000     -0.000     -0.000   
std        0.624      0.606      0.521      0.482      0.404      0.330   
min      -44.808     -2.837    -10.295     -2.605    -22.566    -15.430   
25%       -0.162     -0.355     -0.317     -0.327     -0.071     -0.053   
50%       -0.011      0.041      0.017     -0.052      0.001      0.011   
75%        0.148      0.440      0.351      0.241      0.091      0.078   
max       22.528      4.585      7.520      3.517     31.612     33.848   

          Amount      Class  
count 284807.000 284807.000  
mean      88.350      0.002  
std      250.120      0.042  
min        0.000      0.000  
25%        5.600      0.000  
50%       22.000      0.000  
75%       77.165      0.000  
max    25691.160      1.000  

[8 rows x 31 columns]

In [4]:
df.isnull().sum().max()

0

In [5]:
df['Class'].value_counts()[1]/len(df)*100

0.1727485630620034

We can already observe some things:

- most transactions are quite small, the mean of 'Amount' is only 88 USD
- no nulls in the dataset
- fraudulent transactions are only about 0.17%